Sources/Common/DataModel/PolyLine/index.js

```diff
@@ -0,0 +1,120 @@
+import macro from 'vtk.js/Sources/macros';
+import vtkCell from 'vtk.js/Sources/Common/DataModel/Cell';
+import vtkLine from 'vtk.js/Sources/Common/DataModel/Line';
+
+function vtkPolyLine(publicAPI, model) {
+  model.classHierarchy.push('vtkPolyLine');
+  const superClass = { ...publicAPI };
+
+  model.line = vtkLine.newInstance();
+  model.line.getPoints().setNumberOfPoints(2);
+
+  publicAPI.getCellDimension = () => 1;
+  publicAPI.intersectWithLine = (t1, t2, p1, p2, tol, x, pcoords) => {
+    const outObj = {
+      intersect: 0,
+      t: Number.MAX_VALUE,
+      subId: 0,
+      betweenPoints: null,
+    };
+
+    const numLines = superClass.getPoints().getNumberOfPoints() - 1;
+    let pDistMin = Number.MAX_VALUE;
+    const minXYZ = [0, 0, 0];
+    const minPCoords = [0, 0, 0];
+
+    for (let subId = 0; subId < numLines; subId++) {
+      const pCoords = [0, 0, 0];
+
+      model.line.getPoints().setPoint(0, ...model.points.getPoint(subId));
+      model.line.getPoints().setPoint(1, ...model.points.getPoint(subId + 1));
```

Comment:
The following should be faster:
`model.line.getPoints().getData().set(model.points.getData().subarray(3 * subId, 3 * (subId + 2))`